packages/vat-data/test/test-prepare-revocable.js

```diff
@@ -0,0 +1,127 @@
+// Modeled on test-revoke-heap-classes.js
+
+import { test } from './prepare-test-env-ava.js';
+
+// eslint-disable-next-line import/order
+import { M } from '@agoric/store';
+import { prepareExoClass, prepareExoClassKit } from '../src/exo-utils.js';
+import { makeScalarBigMapStore } from '../src/vat-data-bindings.js';
+import { prepareRevocableKit } from '../src/prepare-revocable.js';
+
+const UpCounterI = M.interface('UpCounter', {
+  incr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+const DownCounterI = M.interface('DownCounter', {
+  decr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+test('test revoke defineVirtualExoClass', t => {
+  const baggage = makeScalarBigMapStore('fakeRootBaggage');
+
+  const makeUnderlyingUpCounter = prepareExoClass(
+    baggage,
+    'UpCounter',
+    UpCounterI,
+    /** @param {number} x */
+    (x = 0) => ({ x }),
+    {
+      incr(y = 1) {
+        const { state } = this;
+        state.x += y;
+        return state.x;
+      },
+    },
+  );
+
+  const makeRevocableUpCounterKit = prepareRevocableKit(
+    baggage,
+    'UpCounter',
+    'UpCounter',
+    ['incr'],
+  );
+
+  const makeUpCounterKit = x =>
+    makeRevocableUpCounterKit(makeUnderlyingUpCounter(x));
+
+  const { revoker, revocable: upCounter } = makeUpCounterKit(3);
+  t.is(upCounter.incr(5), 8);
+  t.is(revoker.revoke(), true);
+  t.throws(() => upCounter.incr(1), {
+    message: '"UpCounter_caretaker" revoked',
```

Comment:
I'm always leery of committing to error messages in our APIs. I tend to do something like...

```suggestion
    message: /revoked/,
```

Reply:
I am conflicted on this point. On one hand, any test of the contents of an error message is a fragile golden, in that the only real requirement is that the diagnostic is helpful, informative, and not misleading. We want to be able to continue to improve error messages on all three of these dimensions, and golden tests put friction in the way. I have experienced this myself, and the friction is non-trivial.

OTOH, the mechanisms creating these supposedly-helpful error messages can be complicated, and the error messages can accidentally become less helpful or more misleading over time without anyone otherwise noticing. So my practice has been towards more complete more golden errors. 

I don't have a good resolution to this tension. Perhaps that practice should change in general. But either way, I'd rather not deviate from my general practice just here.

So, resolving without change. If you disagree, please unresolve this conversation.